### PR TITLE
Add GUI injectTransaction integration test

### PR DESCRIPTION
**Pull request by iketheadore**
Fixes:
- Fix the missing /injectTransaction integration test

Changes:
- Remove .bak wallet file in clean function
- Remove some coin hours checking to reduce the minimal coin hour requirement for full tests
- Split the gui Client.Post into PostForm and PostJSON, the `/injectTransaction` reads raw transaction from json. 

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/api/cli/integration/integration_test.go [hunk 1]
```diff
@@ -1606,7 +1615,7 @@ func TestLiveCreateAndBroadcastRawTransaction(t *testing.T) {
 	}
 
 	// prepares wallet and confirms the wallet has at least 2 coins and 16 coin hours.
```
Comment (therealssj) [inline src/api/cli/integration/integration_test.go]: 2 coin hours

File: src/api/cli/integration/integration_test.go [hunk 2]
```diff
@@ -2184,3 +2144,119 @@ func checkWalletEntriesAndLastSeed(t *testing.T, w *wallet.Wallet) {
...
+
+	doLiveEnvCheck(t)
+
+	if !doWallet(t) {
+		return
+	}
+
+	c := gui.NewClient(nodeAddress())
+	// prepares wallet and confirms the wallet has at least 2 coins and 8 coin hours.
+	w, totalCoins, _ := prepareAndCheckWallet(t, 2e6, 2)
```
Comment (therealssj) [inline src/api/cli/integration/integration_test.go]: check for 8 coin hours

Comment (iketheadore) [reply src/api/cli/integration/integration_test.go]: 2 coin hours are enough, burn 1 in first case, and burn the remaining 1 in second case.

Comment (therealssj) [reply src/api/cli/integration/integration_test.go]: oh true. 
the comment should be updated to say 2 coin hours

Comment (iketheadore) [reply src/api/cli/integration/integration_test.go]: Yes, I will update the comment.

**Pull request merged**